uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestAttachmentTest.php

```diff
@@ -108,36 +110,34 @@ public function testDeleteShouldDeleteAttachmentWhenLoggedInUserIsTheLeaveApprov
   }
 
   public function testDeleteShouldThrowAnExceptionWhenAttachmentHasBeenDeletedBefore() {
-    $adminID = 2;
     $params = $this->getDefaultLeaveRequestParams();
     $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation($params);
 
-    // Register contact in session and set permission to admin
-    $this->registerCurrentLoggedInContactInSession($adminID);
-    CRM_Core_Config::singleton()->userPermissionClass->permissions = ['administer leave and absences'];
+    $leaveManagerService = $this->getLeaveManagerServiceWhenUserIsAdmin();
+    $leaveRequestRights = new LeaveRightsService($leaveManagerService);
+    $leaveRequestAttachmentService = new LeaveRequestAttachmentService($leaveRequestRights, $leaveManagerService);
```

Comment:
This instantiation code is duplicated in a few places. What do you think of moving it to a `createLeaveRequestAttachmentsServiceWhenUserIsAdmin()` (I know, a bit too long, but quite descriptive and still shorter than 3 lines of code)